File: Tests/Full/C#/InputTests.cs
using System;
using System.Collections.Generic;
using System.Text;
using SwinGame;
using System.Drawing;

namespace Tests
{
    class InputTests : IGameTestLoader
    {
        #region IGameTestLoader Members

        public void AddTo(List<TestSuite> list)
        {
            TestSuite result = new TestSuite("Keyboard and Mouse Input Tests");
            result.Add(new InputTest());

            list.Add(result);
        }

        #endregion

        private class InputTest : TestSet
        {
            private readonly static string METHS =
                "All Mouse";

            private readonly static string INST =
                "[H]ide Mouse" + Environment.NewLine +
                "[S]how Mouse" + Environment.NewLine +
                "Move Mouse to [C]enter"  + Environment.NewLine +
                "Click the Left Mouse Button";

            public InputTest() : base(METHS, INST) { }

            protected override void ToRun(System.Drawing.Rectangle toDrawIn)
            {
                Vector MouseMovement = SwinGame.Input.GetMouseMovement();
                
                Text.DrawText("Mouse Position : " + Convert.ToString(SwinGame.Input.GetMousePosition().X) + ", " + Convert.ToString(SwinGame.Input.GetMousePosition().Y), Color.White, GameResources.GameFont("Courier"), 10, 10);
                Text.DrawText("Mouse Movement : " + Convert.ToString(MouseMovement.X) + ", " + Convert.ToString(MouseMovement.Y), Color.White, GameResources.GameFont("Courier"), 10, 30);

                if (SwinGame.Input.WasKeyTyped(Keys.VK_H)) SwinGame.Input.HideMouse();
                if (SwinGame.Input.WasKeyTyped(Keys.VK_S)) SwinGame.Input.ShowMouse();
                if (SwinGame.Input.IsKeyPressed(Keys.VK_C)) SwinGame.Input.MoveMouse(400, 300);

                SwinGame.Graphics.DrawLineOnScreen(Color.LightBlue, Shapes.CreateLine(Input.GetMousePosition().X, 0, Input.GetMousePosition().X, 600));
                SwinGame.Graphics.DrawLineOnScreen(Color.LightBlue, Shapes.CreateLine(0, Input.GetMousePosition().Y, 800,Input.GetMousePosition().Y));

                Text.DrawText("Is Left Mouse Button Down : " + Convert.ToString(SwinGame.Input.IsMouseDown(MouseButton.LeftButton)), Color.White, GameResources.GameFont("Courier"), 10, 50);
                Text.DrawText("Was Left Mouse Button Clicked : " + Convert.ToString(SwinGame.Input.MouseWasClicked(MouseButton.LeftButton)), Color.White, GameResources.GameFont("Courier"), 10, 70);
                Text.DrawText("Is Mouse Cursor Shown : " + Convert.ToString(SwinGame.Input.IsMouseShown()), Color.White, GameResources.GameFont("Courier"), 10, 90);
            }
        }
    }

    class KeyBoardInputTests : IGameTestLoader
    {
        #region IGameTestLoader Members

        public void AddTo(List<TestSuite> list)
        {
            TestSuite result = new TestSuite("Keyboard and Mouse Input Tests");
            result.Add(new KeyBoardInputTest());

            list.Add(result);
        }

        #endregion

        private class KeyBoardInputTest : TestSet
        {
            private readonly static string METHS =
                "All Keyboard";

            private readonly static string INST =
                "Hit the [A] Key";

            public KeyBoardInputTest() : base(METHS, INST) { }

            protected override void ToRun(System.Drawing.Rectangle toDrawIn)
            {
                Text.DrawText("A Key Down : " + Convert.ToString(Input.IsKeyPressed(Keys.VK_A)), Color.White, GameResources.GameFont("Courier"), 10, 10);
                Text.DrawText("A Key Up : " + Convert.ToString(!Input.IsKeyPressed(Keys.VK_A)), Color.White, GameResources.GameFont("Courier"), 10, 30);
                Text.DrawText("A Key Typed : " + Convert.ToString(Input.WasKeyTyped(Keys.VK_A)), Color.White, GameResources.GameFont("Courier"), 10, 50);

                Text.DrawText("Is Reading Text : " + Convert.ToString(Input.IsReadingText()), Color.White, GameResources.GameFont("Courier"), 10, 90);

                //Input.StartReadingText(Color.White, 10, GameResources.GameFont("Courier"), 10, 110);
                //Text.DrawText("You Entered : " + Convert.ToString(Input.TextReadAsASCII()), Color.White, GameResources.GameFont("Courier"), 10, 130);
            }
        }
    }
}

File: Tests/Full/Pascal/InputTests.pas
unit InputTests;

interface
	uses TestFramework;

	procedure AddInputSuite(var suites: TestSuites); 

implementation
	uses GameResources, SGSDK_Input, SGSDK_Core, SGSDK_Graphics, SGSDK_KeyCodes, SGSDK_Shapes, SGSDK_Font, SysUtils;
	
	var
		scroll: Integer;
		MouseMovement: Vector;
		_EnteredText: String;
	
	procedure InputTest(const drawIn: Rectangle);
	begin
        MouseMovement := GetMouseMovement();
       
        DrawText('Mouse Position : ' + FloatToStr(GetMousePosition().X) + ', ' + FloatToStr(GetMousePosition().Y), ColourWhite, GameFont('Courier'), 10, 10);
        DrawText('Mouse Movement : ' + FloatToStr(MouseMovement.X) + ', ' + FloatToStr(MouseMovement.Y), ColourWhite, GameFont('Courier'), 10, 30);

        if WasKeyTyped(VK_H) then HideMouse();
        if WasKeyTyped(VK_S) then ShowMouse();
        if IsKeyPressed(VK_C) then MoveMouse(400, 300);
				if MouseWasClicked(WheelUpButton) then scroll += 1;
				if MouseWasClicked(WheelDownButton) then scroll -= 1;

        DrawLineOnScreen(GetColor(120, 177, 250), CreateLine(GetMousePosition().X, 0, GetMousePosition().X, 600));
        DrawLineOnScreen(GetColor(120, 177, 250), CreateLine(0, GetMousePosition().Y, 800, GetMousePosition().Y));

        DrawText('Is Left Mouse Button Down : ' + BoolToStr(IsMouseDown(LeftButton)), ColourWhite, GameFont('Courier'), 10, 50);
        DrawText('Was Left Mouse Button Clicked : ' + BoolToStr(MouseWasClicked(LeftButton)), ColourWhite, GameFont('Courier'), 10, 70);
        DrawText('Is Mouse Cursor Shown : ' + BoolToStr(IsMouseShown()), ColourWhite, GameFont('Courier'), 10, 90);

				DrawText('Scroll : ' + IntToStr(scroll), ColourWhite, GameFont('Courier'), 10, 140);	
	end;
	
	procedure KeyboardInputTest(const drawIn: Rectangle);
	begin
		DrawText('A Key Down : ' + BoolToStr(IsKeyPressed(VK_A)), ColourWhite, GameFont('Courier'), 10, 10);
		DrawText('A Key Up : ' + BoolToStr(not IsKeyPressed(VK_A)), ColourWhite, GameFont('Courier'), 10, 30);
		DrawText('A Key Typed : ' + BoolToStr(WasKeyTyped(VK_A)), ColourWhite, GameFont('Courier'), 10, 50);

		DrawText('Is Reading Text : ' + BoolToStr(IsReadingText()), ColourWhite, GameFont('Courier'), 10, 90);

		if (not IsReadingText()) and (WasKeyTyped(VK_S)) then
		begin
		    StartReadingText(ColourWhite, 10, GameFont('Courier'), 35, 240);
		end;
		
		if IsReadingText() and MouseWasClicked(LeftButton) then
			_EnteredText := EndReadingText()
		else
			_EnteredText := TextReadAsASCII();
		
		DrawText('You Entered : ' + _EnteredText, ColourWhite, GameFont('Courier'), 10, 130);
	end;
	
	function GetInputTests(): TestSuite;
	var
		i: Integer;
	begin
		result.Title := 'Keyboard and Mouse Input Tests';
		SetLength(result.Tests, 2);
		
		for i := 0 to High(result.Tests) do
		begin
			InitTest(result.Tests[i]);
		end;
		
		with result.Tests[0] do
		begin
			MethodBeingTested := 'All Mouse';
			Instructions := '[H]ide Mouse' + EOL +
	        				'[S]how Mouse' + EOL +
	        				'Move Mouse to [C]enter'  + EOL +
	        				'Click the Left Mouse Button';
			ToRun := @InputTest;
		end;
		
		with result.Tests[1] do
		begin
			MethodBeingTested := 'All Keyboard';
			Instructions := 'Hit the [A] Key' + EOL +
            				'[S]tart Reading Text' + EOL + 
										'Click mouse to end reading text';
			ToRun := @KeyboardInputTest;
		end;
	end;
	
	procedure AddInputSuite(var suites: TestSuites);
	begin
		scroll := 0;
		
		suites[High(suites)] := GetInputTests();
	end;

end.